packages/api-providers/src/web3/vanchor-deposit.ts

```diff
@@ -292,6 +300,11 @@ export class Web3VAnchorDeposit extends VAnchorDeposit<WebbWeb3Provider, Deposit
             message: `${currency.view.name}:wrap and deposit`,
             name: 'Transaction',
           });
+          // TODO: @nepoche set the right leaf index before returning
```

Comment:
The web3 implementation generates the commitment from note secrets and compares to the fetched leaves to determine the index before generating withdraw proofs.